src/ompl/geometric/planners/rrt/RRTstar.h

```diff
@@ -148,42 +160,126 @@ namespace ompl
                 return delayCC_;
             }
 
-            /** \brief Controls whether the tree is pruned during the search. */
-            void setPrune(const bool prune)
+            /** \brief Controls whether the tree is pruned during the search. This pruning removes
+                a vertex if and only if it \e and all its descendents passes the pruning condition.
+                The pruning condition is whether the lower-bounding estimate of a solution
+                constrained to pass the the \e vertex is greater than the current solution.
+                Considering the descendents of a vertex prevents removing a descendent
+                that may actually be capable of later providing a better solution once
+                its incoming path passes through a different vertex (e.g., a change in homotopy class). */
+            void setTreePruning(const bool prune)
```

Comment:
So I realised that half the point of this to be able to propose changes and easily compare them. I've rewritten the pruning function to be less aggressive. Open to discussion if that's what you want.